state/protocol/inmem/convert.go

```diff
@@ -0,0 +1,188 @@
+package inmem
+
+import (
+	"errors"
+
+	"github.com/onflow/flow-go/model/encodable"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/state/protocol"
+)
+
+// FromSnapshot generates a memory-backed snapshot from the input snapshot.
+// Typically, this would be used to convert a database-backed snapshot to
+// one that can easily be serialized to disk or to network.
+func FromSnapshot(from protocol.Snapshot) (*Snapshot, error) {
+
+	var (
+		snap EncodableSnapshot
+		err  error
+	)
+
+	// convert top-level fields
+	snap.Head, err = from.Head()
+	if err != nil {
+		return nil, err
+	}
+	snap.Identities, err = from.Identities(filter.Any)
+	if err != nil {
```

Comment:
I would add some prefix to this error letting them get nested: `return nil, fmt.Errorf("could not retrieve identities: %w")`

Reply:
Same suggestion for the rest of the errors (unless you want to preserve their type).